storage/mysql/tree_storage.go

```
@@ -404,6 +404,13 @@ func (t *treeTX) Rollback() error {
 	return nil
 }
 
+func (t *treeTX) Close() error {
+	if !t.closed {
+		return t.Rollback()
```

All of the deferred Close() calls are hiding errors if the rollback fails.